exasol_advanced_analytics_framework/udf_framework/dynamic_modules.py

```diff
@@ -0,0 +1,24 @@
+import sys
+import importlib
+from typing import Any
+from types import ModuleType
+
+
+def create_module(name: str) -> ModuleType:
+    """
+    Dynamically create a python module using the specified name and
+    register the module in sys.modules[].
+
+    Additionally add a function add_to_module() to the module enabling other
+    code to add classes and functions to the module.
+    """
+    spec = importlib.machinery.ModuleSpec(name, None)
+    mod = importlib.util.module_from_spec(spec)
+    sys.modules[name] = mod
```

Comment:
extract to _register_module_for_import

Reply:
how do we handle the case if the module already exists?

Reply:
See next push 
* moving operations to separate functions `_create_module()` and `_register_module_for_import()`
* Covering the case if the module already exists
* Enhancing unit tests to verify the case if the module already exists